PyInstaller/building/api.py

```diff
@@ -61,16 +61,19 @@ def __init__(self, *tocs, **kwargs):
 
             name
                 A filename for the .pyz. Normally not needed, as the generated name will do fine.
-            cipher
-                The block cipher that will be used to encrypt Python bytecode.
         """
+        if kwargs.get("cipher"):
+            raise SystemExit(
+                "The 'cipher' option (i.e. bytecode encryption feature) has been removed. Please "
+                "remove the 'cipher' arguments to PYZ() and Analysis() in your spec file. "
+                "For the rational and alternatives see https://github.com/pyinstaller/pyinstaller/pull/6999"
```

Comment:
rationale

Reply:
Also, this exception/message seems to be repeated in several places. Would it make sense to put (at least the common part of it) into a constant? Or maybe have a custom exception class that allows custom message + hard-coded common part?